packages/@okta/vuepress-site/docs/guides/terraform-syntax-tips/main/index.md

```diff
@@ -0,0 +1,201 @@
+
+## Reduce code duplication with loops
+
+Consider using a Terraform loop when your code assigns different values to the same kind of resource, such as creating multiple groups. The Terraform meta-argument `for_each` iterates over multiple values that you specify. You can perform the same statements using each value in the set. See [the Terraform documentation](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each) for this feature.
+
+Using a loop enables you to reduce duplication of your Terraform code. This approach makes your Okta Terraform code readable, maintainable, and less prone to deployment errors and inconsistencies.
+
+Within a resource definition, set the `for_each` meta-argument to a set or map of your list-like data.
+
+After assigning the `for_each` meta-argument, use the special phrase `each.value` to represent the current item in the loop across the input map or set. Set properties on each object, such as `each.value.name`.
+
+> **Note**: Attribute values tagged as sensitive aren't permitted as arguments to `for_each`. See [Store secrets and credentials securely](/docs/guides/terraform-organize-configuration/main/#store-secrets-and-credentials-securely).
+
+This example creates groups from a list of group names:
+
+1. In your Terraform file for groups, define a variable for your group definitions. For this example, assume this is in a file called `users_and_groups.tf`:
+
+    ```hcl
+    variable "groups_to_create" {
+      description = "List of group objects"
+      type = list(object({
+        name = string
+        description = string
+       }))
+    }
+    ```
+
+1. Create a file for variable values, in this case, a list of group objects to create. If you aren't using the central `terraform.tfvars` file to set variable values, name the file with the suffix `.auto.tfvars`. For example, `users_and_groups.auto.tfvars`.
+
+   Add this example to the file:
+
+    ```hcl
+    groups_to_create = [
+      {
+        "name": "Group 1",
+        "description": "Group 1 description"
+      },
+      {
+        "name": "Group 2",
+        "description": "Group 2 description"
+      },
+      {
+        "name": "Group 3",
+        "description": "Group 3 description"
+      }
+    ]
+    ```
+
+1. In your original Terraform `.tf` file, create a group resource definition:
+
+    ```hcl
+    resource "okta_group" "group_create_using_list" {
+
+    }
+    ```
+
+1. In that resource definition, set `for_each` to a map or set. You can convert array content to a map before setting the meta-argument:
+
+    ```hcl
+    for_each = { for group in var<your-variable-name>
+      : group.name => group}
+    ```
+
+   Using the previous examples:
```

Comment:
I don't understand what "using the previous examples" means. Also, can you make it a complete sentence in case we translate this page eventually? (stem sentences break translations)

Reply:
Nice catch... unfortunately I'm only cleaning this up for a contract writer... however, here's the replacement that I belive makes more sense:

>     ```hcl
>     for_each = { for group in var.{yourGroupVariableName}
>       : group.name => group}
>     ```
> 
>    Using the `groups_to_create` variable created earlier, this becomes:
> 
>     ```hcl
>     for_each = { for group in var.groups_to_create
>       : group.name => group}
>     ```
>